haystack/components/preprocessors/recursive_chunker.py

```diff
@@ -0,0 +1,212 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import re
+from typing import Any, Dict, List
+
+from haystack import Document, component, default_from_dict, default_to_dict, logging
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class RecursiveChunker:
+    """
+    Recursively chunk text into smaller chunks.
+
+    This component is used to split text into smaller chunks, it does so by recursively applying a list of separators
+    to the text.
+
+    Each separator is applied to the text, if then checks each of the resulting chunks, it keeps the ones chunks that
+    are within the chunk_size, for the ones that are larger than the chunk_size, it applies the next separator in the
+    list to the remaining text.
+
+    This is done until all chunks are smaller than the chunk_size parameter.
+
+    Example:
+
+    ```python
+    from haystack import Document
+    from haystack.components.preprocessors.recursive_chunker import RecursiveChunker
```

Comment:
RecursiveChunker (or RecursiveDocumentSplitter) should be added to haystack/components/preprocessors/__init__.py and then you can simplify this import path to `from haystack.components.preprocessors import ...`